TIL/daily/2025-06-13.md

```
@@ -0,0 +1,18 @@
+  ### 4. 테스트 시 상속에 관련된 테스트는 의미가 없음으로 기능테스트가 아닌 이상 실행하지 않는다.
+  ### 5. 테스트 Description을 작성할 때는 AI를 통해 1차 검증해서 구체적인 테스트 케이스를 설계해도 괜찮다.
+  ### 6. 테스트 시 반복되는 코드의 경우 메서드화해서 처리한다.
+  ### 7. if-else 문의 흐름에 따라 모든 경우의 수를 체크해야한다.
+
+## Abstract Class(추상 클래스)
+  ### 1. 추상 클래스는 미정의된 메서드를 작성하여 뼈대를 설계하는 느낌인것 같다.
+  ### 2. 추상 클래스와 interface의 차이를 이해하지 못했는데 추상 클래스의 경우 생성자를 만들지 못하고 추상 메서드를 선언하거나 아닌 메서드도 가질 수 있다는 사실을 알았다.
+  ### 3. 추상 클래스는 미정의된 것들을 가질 수 있으니까 implements를 해도 override를 할 필요는 없다.
+
+## Interface(인터페이스)
+  ### 1. 인터페이스의 경우 모든 메서드가 추상메서드여야 한다. 이부분에서 추상 클래스와 기존에 헷갈렸는데 명확하게 구분지을 수 있게됨.
```

markdownlint-cli2 (0.17.2)

Headings must start at the beginning of the line
null

(MD023, heading-start-left)